notebooks/reference_scenarios/benchmarks.ipynb

```diff
@@ -0,0 +1,282 @@
+    "                samples_split_option='stratified',\n",
+    "                multi_partner_learning_approach='fedavg', #default\n",
+    "                aggregation='data-volume', #default\n",
+    "                contributivity_methods=[\"Independent scores\",\"Shapley values\"],\n",
+    "                epoch_count=NB_EPOCH,\n",
+    "                minibatch_count = NB_MINIBATCH,\n",
+    "                gradient_updates_per_pass_count=NB_GRAD_UPDATE,\n",
+    "            )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### #2 | A different duo: Good cop / Bad cop [CIFAR10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Random split of data among 2 partners (70% / 30%). Half of the 2nd partner’s data is corrupted."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc2 = Scenario(\n",
+    "                scenario_id = 2,\n",
+    "                partners_count=2,\n",
+    "                dataset='cifar10',\n",
+    "                amounts_per_partner = [0.7, 0.3],\n",
+    "                samples_split_option='random', #default\n",
+    "                corruption_parameters=['not-corrupted',Randomize(proportion=0.5)],\n",
+    "                multi_partner_learning_approach='fedavg', #default\n",
+    "                aggregation='data-volume', #default\n",
+    "                contributivity_methods=[\"Independent scores\",\"Shapley values\"],\n",
+    "                epoch_count=NB_EPOCH,\n",
+    "                minibatch_count = NB_MINIBATCH,\n",
+    "                gradient_updates_per_pass_count=NB_GRAD_UPDATE,\n",
+    "            )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### #3 | A trio: Two cool guys and their challenger [CIFAR10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Mixed split of data among 3 partners (42%, 42%, 16%). Samples of 3 classes are randomly split, cool guys get 3 specific classes each, challenger only 1."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sc3 = Scenario(\n",
+    "                scenario_id=3,\n",
+    "                partners_count=3,\n",
+    "                dataset='cifar10',\n",
+    "                amounts_per_partner=[0.42, 0.42, 0.16],\n",
+    "                samples_split_option='flexible',\n",
+    "                samples_split_configuration=[[0.4, 0.4, 0.4, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0], \n",
+    "                                                [0.4, 0.4, 0.4, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 0.0],\n",
+    "                                                [0.2, 0.2, 0.2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0],\n",
+    "                                                ],\n",
```

Comment:
When the flexible splitter is used, the `amounts_per_partner` are discarded and inferred from `samples_split_configuration`.
So for the first three classes it should be 0.42, 0.42, 0.16 instead of 0.4, 0.4, 0.2.

Reply:
You meant it should be 0.4, 0.4, 0.2 instead of 0.42, 0.42, 0.16 ?

Reply:
No I mean if we stick to the description of the 3rd ref scenario, the amounts of data per partner should be 0.42, 0.42, 0.16. So as with the `FlexibleSplitter` the amounts are inferred from `samples_split_configuration`, we should have in that variable, for the first three classes, 0.42, 0.42 and 0.16. No? 😃

Reply:
Maybe I misunderstood but especially because the amounts of data per partner should be 0.42, 0.42, 0.16, the first three classes should be 0.4 for partner #1 for example to get 0.4+0.4+0.4+1+1+1=4.2
Maybe I am missing something here 😃

Reply:
There is a confusion indeed 😄 I should improve the documentation clarity. So, let me explain:

- 0.42, 0.42, 0.16 are relative amounts of data, i.e. 42% 42% 16% of the whole train dataset
- For samples with classes 4 to 10, the split is 3 whole classes - 3 whole classes - 1 whole class. Which gives: 3/7th, 3/7th, 1/7th, i.e. 42% 42% 16%
- Now for the first 3 classes, we have to enforce also 42% 42% 16%. So as we share those ones among the 3 partners, we give respectively 42% 42% 16% of each of them

Does this clarify?